exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TraceExporterTest.java

```diff
@@ -15,25 +15,129 @@
  */
 package com.google.cloud.opentelemetry.trace;
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.trace.v2.TraceServiceClient;
+import com.google.cloud.trace.v2.TraceServiceSettings;
+import com.google.cloud.trace.v2.stub.TraceServiceStub;
+import com.google.devtools.cloudtrace.v2.ProjectName;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
 import io.opentelemetry.sdk.trace.export.SpanExporter;
 import java.io.IOException;
+import java.util.Collections;
+import org.junit.After;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
 
-@RunWith(JUnit4.class)
+@RunWith(MockitoJUnitRunner.class)
 public class TraceExporterTest {
 
+  private static final String PROJECT_ID = "test-id";
+  @Mock private TraceServiceClient mockedTraceServiceClient;
+  @Mock private TraceServiceStub mockedTraceServiceStub;
+
+  @After
+  public void tearDown() {
+    GlobalOpenTelemetry.resetForTest();
+  }
+
+  @SuppressWarnings("ResultOfMethodCallIgnored")
   @Test
   public void createWithConfiguration() {
```

Comment:
This is no longer just a "will the constructor work" unit test.  You're fully testing to a  more integration-level here.

As such I'd update the test name to denote what it's doing *or* limit what you test to just what was done previously.

Reply:
Right, this now verifies that export is working as well. I will rename this to `verifyExporterWorksWithConfiguration` .